node/src/components/storage.rs

```diff
@@ -1141,6 +1141,17 @@ impl Storage {
             .transpose()
     }
 
+    /// Retrieves the highest block header from the storage, if one exists.
+    pub fn read_highest_block_header(&self) -> Result<Option<BlockHeader>, Error> {
+        let highest_block_hash = match self.block_height_index
+            .iter()
+            .last() {
```

Comment:
Seems like having a separate value just for tracking the highest block would be useful. I can imagine a query like `GetHighestBlock`.

Reply:
I don't think it needs a separate field: Accessing the last entry of a `BTreeMap` is fast.

Reply:
Ok.